### PR TITLE
Path tests overhaul

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Path tests weren't running due to a test infra or runtime bug (too many test methods in one class apparently). Cleaned up path tests, fixing issues and adding coverage.

cc: @danmosemsft, @Anipik, @maryamariyan, @pjanotti 

See https://github.com/dotnet/coreclr/pull/16554, #27426

### CODE REVIEW

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.Windows.cs [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+        [Theory,
+            InlineData("B:", ""),
+            InlineData("A:.", ".")]
+        public static void GetFileName_Volume(string path, string expected)
+        {
+            // With a valid drive letter followed by a colon, we have a root, but only on Windows.
+            Assert.Equal(expected, Path.GetFileName(path));
+        }
+
+        [ActiveIssue(27269, TargetFrameworkMonikers.Netcoreapp)]
```
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.Windows.cs]: did we get this fix yet?

Comment (JeremyKuhne) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests.Windows.cs]: No, unfortunately. I'm not sure why we don't have a new CoreLib, I've run out of mental bandwidth to figure it out. Once we do, we can just remove these few and everything should pass. Well, with https://github.com/dotnet/coreclr/pull/16554 anyway. :)

**Comment by danmosemsft** [comment]
CoreCLR build is broken so no insertion. https://github.com/dotnet/core-eng/issues/2753

**Comment by jkotas** [comment]
I do not think we should be working around the bug by refactoring the tests. The bug is likely affecting random other places. We need to get that bug traced down and fixed fast.

**Comment by jkotas** [comment]
E.g. I see that `System.Tests.EnvironmentTests` are effected by this as well - some of them are getting skipped randomly.

**Comment by JeremyKuhne** [comment]
> I do not think we should be working around the bug by refactoring the tests.

I needed to fix the tests anyway- they were broken and scenarios were missing/incorrect. It was an 8+ hour effort to get them corrected for both the currently ingested bits and the pending CoreCLR changes from the past few days. They are also better factored for me to make additional passes to beef up the coverage.

It is good we now know that we've got a runtime problem and it is reproing in multiple ways. Should we track with #27426 or open another issue in CoreCLR?

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.netcoreapp.cs [hunk 2]
```diff
@@ -1,4 +1,3 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (jkotas) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.netcoreapp.cs]: Unintentional edit

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.Windows.netcoreapp.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public partial class PathTests_Windows : PathTestsBase
```
Comment (jkotas) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.Windows.netcoreapp.cs]: If this refactoring is perfmanent, the name of the file should be `PathTests_Windows` to follow the naming conventions.

File: src/System.Runtime.Extensions/tests/TestHelpers.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System
+{
+    public static class TestHelpers
+    {
+
```
Comment (jkotas) [inline src/System.Runtime.Extensions/tests/TestHelpers.cs]: Extra line

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.Windows.netcoreapp.cs (hunk 3 above)
Comment (jkotas) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests.Windows.netcoreapp.cs]: (Similar for other places.)

File: src/System.Runtime.Extensions/tests/System/IO/Path.Combine.cs [hunk 5]
```diff
@@ -7,7 +7,7 @@
 
 namespace System.IO.Tests
 {
-    public static partial class PathTests
+    public class PathTests_Combine
```
Comment (jkotas) [inline src/System.Runtime.Extensions/tests/System/IO/Path.Combine.cs]: Nit: file name vs. class name mismatch

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Windows.netfx.cs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    [PlatformSpecific(TestPlatforms.Windows)]
+    [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)]
+    public class PathTests_NetFX : PathTestsBase
```
Comment (jkotas) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests_Windows.netfx.cs]: Nit: File name vs class name mismatch

Comment (JeremyKuhne) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests_Windows.netfx.cs]: This one is a bit awkward. I really want to keep it in the build as it makes it much easier to work with, but I don't want to litter the file with [SkipOnTargetFramework] and yet still clearly show the relationship when browsing the solution explorer. :/

**Comment by JeremyKuhne** [comment]
@dotnet/dncenghot It appears that the Ubuntu.1804.Amd64.Open runs are hung up? All of the Linux x64 Release Builds are spinning since last night.

**Comment by danmosemsft** [comment]
1804 is jammed on other PR's also.

**Comment by JeremyKuhne** [comment]
@danmosemsft Should we skip on 1804 if all other leg results are good?

**Comment by danmosemsft** [comment]
@jeremykuhne let's see whether it can be quickly fixed tomorrow morning.

**Comment by JeremyKuhne** [comment]
> let's see whether it can be quickly fixed tomorrow morning.

We don't have much time left in the month. :/ Note that this one will be necessary to pass the next CoreCLR ingestion attempt.

**Comment by danmosemsft** [comment]
Ugh..OK, skipping it here seems reasonable given that fact and that is new and we have other Ubuntu coverage.

**Comment by JeremyKuhne** [comment]
> skipping it here seems reasonable

Thanks :)

**Comment by MattGal** [comment]
@JeremyKuhne  looking at the Ubuntu 18.04 queue now.  I think the problem is simply that the images expire faster than we'd like as it's not RTM yet.

**Comment by MattGal** [comment]
@JeremyKuhne & @danmosemsft :  This is fixed now.

_Summary_: Since Ubuntu 18 is not RTMed  yet we run a daily build, and they update their image VERY often.  I've moved our VMs to set to "latest" version (which has caused problems in the past when latest = broken build) until a release version 18.04 is finalized to prevent this going forward.

**Comment by danmosemsft** [comment]
Thanks @MattGal !

**Pull request merged**